api/lib/domain/services/authentication/oidc-authentication-service.js

```diff
@@ -116,35 +117,42 @@ class OidcAuthenticationService {
     if (!response.isSuccessful) {
       const message = 'Une erreur est survenue en récupérant les informations des utilisateurs.';
       const dataToLog = httpErrorsHelper.serializeHttpErrorResponse(response, message);
-
       monitoringTools.logErrorWithCorrelationIds({ message: dataToLog });
-
-      throw new InvalidExternalAPIResponseError(
-        'Une erreur est survenue en récupérant les informations des utilisateurs.'
-      );
+      throw new InvalidExternalAPIResponseError(message);
     }
 
     const userInfoContent = response.data;
 
     if (!userInfoContent || typeof userInfoContent !== 'object') {
-      const message = 'Les informations utilisateur récupérées ne sont pas au format attendu.';
+      const message =
+        "Les informations utilisateur renvoyées par votre fournisseur d'identité ne sont pas au format attendu.";
       const dataToLog = {
         message,
         typeOfUserInfoContent: typeof userInfoContent,
         userInfoContent,
       };
       monitoringTools.logErrorWithCorrelationIds({ message: dataToLog });
-      throw new InvalidExternalAPIResponseError(message);
+      const error = OIDC_ERRORS.USER_INFO.badResponseFormat;
+      const meta = {
+        shortCode: error.shortCode,
+      };
+      throw new OidcUserInfoFormatError(message, error.code, meta);
     }
 
     const userInfoMissingFields = this.userInfoMissingFields({ userInfoContent });
```

Comment:
Même si ce n'est pas cette PR qui est à l'origine du nom de la méthode, on gagnerait en lisibilité du code à renommer dans cette PR le nom de la méthode `userInfoMissingFields` en `getUserInfoMissingFields` qui commence par un verbe :
```suggestion
    const userInfoMissingFields = this.getUserInfoMissingFields({ userInfoContent });
```

Reply:
> le service OIDC du fournisseur d'accès était manifestement en erreur et que cela touchait tous les utilisateurs

Non pas tous les utilisateurs, seulement ceux qui quand on décode le token renvoyé par pole emploi, n'ont pas les élements requis (family name, etc.)
Vu que l'appel à userInfo n'est fait par défaut mais seulement dans ce cas précis (qui ne touche que pole emploi il me semble).
J'ai pas d'avis sur la question.